www/templates/intro/survey.html

```diff
@@ -1,5 +1,5 @@
 <ion-content class="has-footer">
-    <tutorial/>
+    <tutorial test="surveyState.test" isValid="surveyState.isValid" schema="surveyState.schema" result="surveyState.result"/>
```

Comment:
Not sure why you are passing the individual fields. can you just pass in the `surveyState` object directly?
Probably best to try that first, since we want to pass in the schema object with sub-fields next.

Reply:
I am attempting to pass the survey state via: 
`surveyState="surveyState"` 
but I am hitting a hurdle with then accessing the variables within the $scope.

Reply:
https://stackoverflow.com/questions/49081549/passing-object-as-props-to-jsx

Thought something like this would work.. alas I can't seem to get it right.

Reply:
what do you mean by accessing variables within the $scope?
Also, you are already passing in an object (`surveyState.schema`) and it looked like that worked?
Does it mean that top level objects directly under `$scope` don't work?

Reply:
Within the .jsx file, we do the prop type checking. I believe one of the issues here is that we originally passed in the name, ie isValid="isValid". Now, we are only passing in the $scope variable surveyState. Code from .jsx:
```
propTypes: {
        isValid: React.PropTypes.bool,
        schema: React.PropTypes.object,
        result: React.PropTypes.object
    }
```
My initial instinct was to attempt something like surveyState.result: `React.PropTypes.object`, but this did not work. 

A few ways I've tried to pass in the surveyState: 
`<tutorial {surveyState} />`
`<tutorial {...surveyState} />`
`<tutorial surveyState="surevyState" />`

Reply:
well, you would need to change the propTypes to accept a `surveyState` object as well

Also, check the spelling on ` <tutorial surveyState="surevyState" />`

Reply:
If we changed the propTypes to accept a surveyState, would that be the only check? 
ie. would we have something like: 
```
propTypes : {
    surveyState: React.propTypes.object 
} 
```
I am not seeing how the propTypes would be changed to accept a surveyState object exactly. We would still want to do all the confirmation that isValid is a boolean etc.

Reply:
yeah that would be the only check. Note that we already don't do any type checking within either the schema or the results, so we only lose type information for `isValid` with this approach.

Let's first see if it works, and then we can make the design decision.

Reply:
Okay, here is a commit that does not quite work yet, but seems to be what we want. Still trying to figure out why this would not work. https://github.com/shankari/e-mission-phone/pull/4/commits/d48bba1f590b346b215d68726d2a8a8977046d38